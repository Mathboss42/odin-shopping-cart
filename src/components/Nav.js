import { Link } from 'react-router-dom';

function Nav(props) {
    const navItems = ['HOME', 'STORE']

    return (
        <div className='nav'>
            <h1>CyberShop<span className='small-number'>2023</span></h1>
            <ul>
                {navItems.map((el, index) => {
                    return (
                        <li className={props.selected.toLowerCase() === el.toLocaleLowerCase() ? 'selected' : null} key={index}>
                            <Link to={`/${el.toLowerCase()}`}>{el}</Link>
                        </li>
                    );
                })}
            </ul>
        </div>
    );
}

export default Nav;