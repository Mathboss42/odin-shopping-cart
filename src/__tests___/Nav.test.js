import React from "react";
import userEvent from "@testing-library/user-event";
import { render } from "@testing-library/react";
import Nav from "../components/Nav";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "@remix-run/router";
import { screen } from "@testing-library/react";

describe('Nav Routing', () => {
    it('should route from home to store when clicking store link', () => {
        const history = createMemoryHistory({ initialEntries: ['/home'] });
        render(
            <Router location={history.location} navigator={history}>
                <Nav selected="Home" />
            </Router>
        );
        expect(history.location.pathname).toBe('/home');
        userEvent.click(screen.getByText(/store/i));
        expect(history.location.pathname).toBe('/store');
    });

    it('should route from store to home when clicking home link', () => {
        const history = createMemoryHistory({ initialEntries: ['/store'] });
        render(
            <Router location={history.location} navigator={history}>
                <Nav selected="store" />
            </Router>
        );
        expect(history.location.pathname).toBe('/store');
        userEvent.click(screen.getByText(/home/i));
        expect(history.location.pathname).toBe('/home');
    });
});

describe('Nav Structure', () => {
    it('Should display heading and links', () => {
        const history = createMemoryHistory({ initialEntries: ['/store'] });
        const { container } = render(
            <Router location={history.location} navigator={history}>
                <Nav selected="store" />
            </Router>
        );
        expect(container).toMatchSnapshot();
    });
});