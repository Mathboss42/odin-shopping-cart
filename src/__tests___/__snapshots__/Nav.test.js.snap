// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Structure Should display heading and links 1`] = `
<div>
  <div
    class="nav"
  >
    <h1>
      CyberShop
      <span
        class="small-number"
      >
        2023
      </span>
    </h1>
    <ul>
      <li>
        <a
          href="/home"
        >
          HOME
        </a>
      </li>
      <li
        class="selected"
      >
        <a
          href="/store"
        >
          STORE
        </a>
      </li>
    </ul>
  </div>
</div>
`;
